saleor/app/manifest_validations.py

```diff
@@ -349,3 +360,30 @@ def clean_author(author) -> Optional[str]:
     raise ValidationError(
         "Incorrect value for field: author", code=AppErrorCode.INVALID.value
     )
+
+
+def fetch_icon_image(url: str, field_name: str) -> ContentFile:
+    filename = get_filename_from_url(url)
+    try:
+        with requests.get(url, stream=True, timeout=REQUEST_TIMEOUT) as res:
```

Comment:
What do you think about fetching this image async in a task with some retries if the initial request doesn't work? I'm just thinking that maybe the logo can be loaded later and doesn't have to slow down the app installation process. I won't argue about that; it's just an idea.

Reply:
Yeah, I am still thinking that logo is something optional, and if we can't fetch it, we should just allow to install the app and maybe try fetching it once again.